docs/guides/aws-private-link-workspace.md

```diff
@@ -0,0 +1,246 @@
+  - Two AWS VPC endpoints
+- Workspace Creation
+
+## Provider initialization
+
+Initialize [provider with `mws` alias](https://www.terraform.io/language/providers/configuration#alias-multiple-provider-configurations) to set up account-level resources. See [provider authentication](../index.md#authenticating-with-hostname,-username,-and-password) for more details.
+
+```hcl
+terraform {
+ required_providers {
+   databricks = {
+     source  = "databrickslabs/databricks"
+     version = "0.4.7"
+   }
+   aws = {
+     source = "hashicorp/aws"
+     version = "3.49.0"
+   }
+ }
+}
+
+provider "aws" {
+ region = var.region
+}
+
+// initialize provider in "MWS" mode for provisioning workspace with AWS PrivateLink
+provider "databricks" {
+ alias    = "mws"
+ host     = "https://accounts.cloud.databricks.com"
+ username = var.databricks_account_username
+ password = var.databricks_account_password
+}
+```
+
+Define the required variables
+
+```hcl
+variable databricks_account_id {}
+variable databricks_account_username {}
+variable databricks_account_password {}
+variable vpc_id {}
+variable network_name {}
+variable region {}
+variable existing_network_sg {}
+variable existing_network_subnets {}
+
+variable workspace_vpce_service {}
+variable relay_vpce_service {}
+
+variable vpc_cidr_block {}
+variable vpce_cidr {}
+
+// Use the Databricks Account API 2.0 to retrieve these two IDs below - https://docs.databricks.com/dev-tools/api/latest/account.html
+variable credentials_id {}
+variable storage_configuration_id {}
+
+locals {
+ prefix = "private-link-ws"
+}
+
+locals {
+workspace_subnet_1 = (
+ element(split(",", var.existing_network_subnets), 1)
+)
+workspace_subnet_2 = (
+ element(split(",", var.existing_network_subnets), 2)
+)
+}
+```
+
+## Configure AWS objects
+The first step is to create the required AWS objects:
+- A subnet dedicated to your VPC endpoints
+- A security group dedicated to your VPC endpoints and satisfying required inbound/outbound TCP/HTTPS traffic rules on ports 443 and 6666, respectively.
+- Lastly, creation of the private access settings and workspace.
+
+```hcl
+resource "aws_subnet" "vpce" {
+ vpc_id     = var.vpc_id
+ cidr_block = var.vpce_cidr
+
+ tags = {
+   Name = "vpce subnet for workspace"
+ }
+}
+```
+
+```hcl
+resource "aws_security_group" "vpce_sg" {
+ name        = "VPC endpoint security group"
+ description = "Security group shared with relay and workspace endpoints"
+ vpc_id      = var.vpc_id
+
+ ingress {
+   description      = "Inbound rules"
+   from_port        = 443
+   to_port          = 443
+   protocol         = "tcp"
+   cidr_blocks      = [var.vpc_cidr_block]
+ }
+
+ ingress {
+   description      = "Outbound rules"
+   from_port        = 6666
+   to_port          = 6666
+   protocol         = "tcp"
+   cidr_blocks      = [var.vpc_cidr_block]
+ }
+
+ egress {
+   from_port        = 0
+   to_port          = 0
+   protocol         = "-1"
+   cidr_blocks      = ["0.0.0.0/0"]
```

Comment:
I think it is a good practice to limit the protocol and the ports for egress if possible. https://docs.databricks.com/administration-guide/cloud-configurations/aws/customer-managed-vpc.html#security-groups

Reply:
Done.